packages/ckb/src/utils/ckb-tx.ts

```diff
@@ -234,3 +245,58 @@ export const checkCkbTxInputsCapacitySufficient = async (
     .reduce((prev, current) => prev + current, BigInt(0));
   return sumInputsCapacity > sumOutputsCapacity;
 };
+
+export function signCkbTransaction(
+  key: string,
+  ckbTx: CKBComponents.RawTransactionToSign,
+  inputCells?: { outPoint: CKBComponents.OutPoint; lock: CKBComponents.Script }[],
+  skipMissingKeys?: boolean,
+): CKBComponents.RawTransaction;
+
+export function signCkbTransaction(
+  key: Map<string, string>,
+  ckbTx: CKBComponents.RawTransactionToSign,
+  inputCells: { outPoint: CKBComponents.OutPoint; lock: CKBComponents.Script }[],
+  skipMissingKeys?: boolean,
+): CKBComponents.RawTransaction;
+
+export function signCkbTransaction(
```

Comment:
Why are there three functions with the same name `signCkbTransaction`?

Reply:
The first overload handles the case where the key is a single string and `inputCells` is optional. The second overload is for when key is a `Map<string, string>`, in which case `inputCells` is required. The third definition is the actual implementation. I believe these overloads improve type checking and IDE support by explicitly defining valid parameter combinations, even though they do add some complexity.

Reply:
Can you verify if it works if there is only one function signature?

Reply:
Yes, the third function, which is the actual implementation, is sufficient on its own. The following are the related transactions I just sent for xUDT testing on Testnet3 in offline mode:
- Issuance preparation
  - CKB tx: https://testnet.explorer.nervos.org/transaction/0x922a0e9ef610a1d58af1541025c3269001cd1c2f1e08d5b0464f86ad5d5fd7ca
- Distribution
  - BTC tx: https://mempool.space/testnet/tx/e6d5d814bebe1aac6c045297032f92dc62e3d6e24757967e0e60b77e890de12c
  - CKB tx: https://testnet.explorer.nervos.org/transaction/0x200eaf55e2b1e231216cf297caa70c5c26308fd47f25bcd881c081efdaf97061

Reply:
So can we remove the above two functions?

Reply:
Sure, I'll make the change.

Reply:
Addressed in 740612b80e9a86aa8df8c53a50e462d65dbd1842, along with this suggested change https://github.com/utxostack/rgbpp-sdk/pull/302#discussion_r1948957389.